pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -994,11 +995,17 @@ async def _map_user_message(self, message: ModelRequest) -> AsyncIterable[chat.C
             elif isinstance(part, UserPromptPart):
                 yield await self._map_user_prompt(part)
             elif isinstance(part, ToolReturnPart):
+                response_parts = part.model_response_parts()
+                multimodal_parts = [p for p in response_parts if isinstance(p, (FileUrl, BinaryContent))]
```

Comment:
See above; I'd rather not repeat this but cleanly split up the fields/methods on `UserPromptPart`